lib/core/presentation/pages/chat/chat_message/widgets/message_item/message_reaction_widget.dart

```diff
@@ -0,0 +1,228 @@
+    final colorScheme = Theme.of(context).colorScheme;
+    final reactions = getReactions();
+    return Wrap(
+      spacing: 4,
+      runSpacing: 4,
+      crossAxisAlignment: WrapCrossAlignment.center,
+      children: [
+        ...reactions
+            .map(
+              (r) => _Reaction(
+                reactionKey: r.key,
+                count: r.count,
+                reacted: r.reacted,
+                onTap: () => _onTapReaction(r),
+                onLongPress: () async => await _AdaptableReactorsDialog(
+                  reactionEntry: r,
+                ).show(context),
+              ),
+            )
+            .toList(),
+        if (reactions.isNotEmpty)
+          InkWell(
+            onTap: () {
+              BottomSheetUtils.showSnapBottomSheet(context, builder: (_) {
+                return LemonEmojiPicker(
+                  onEmojiSelected: (emoji) {
+                    Navigator.of(context).pop();
+                    onReact?.call(
+                      event: event,
+                      emoji: emoji.emoji,
+                    );
+                  },
+                );
+              });
+            },
+            borderRadius: BorderRadius.circular(LemonRadius.small),
+            child: Container(
+              decoration: BoxDecoration(
+                color: colorScheme.surface,
+                borderRadius: BorderRadius.circular(LemonRadius.small),
+              ),
+              padding: EdgeInsets.symmetric(
+                horizontal: Spacing.superExtraSmall,
+                vertical: Spacing.superExtraSmall / 2,
+              ),
+              child: Icon(
+                Icons.add_circle_outline,
+                color: colorScheme.onSurface,
+                size: Sizing.xSmall,
+              ),
+            ),
+          ),
+      ],
+    );
+  }
+}
+
+class _Reaction extends StatelessWidget {
+  final String? reactionKey;
+  final int? count;
+  final bool? reacted;
+  final void Function()? onTap;
+  final void Function()? onLongPress;
+
+  const _Reaction({
+    this.reactionKey,
+    this.count,
+    this.reacted,
+    this.onTap,
+    this.onLongPress,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final color = reacted == true ? LemonColor.lavender : Theme.of(context).colorScheme.surface;
+
+    return InkWell(
+      onTap: () => onTap != null ? onTap!() : null,
+      onLongPress: () => onLongPress != null ? onLongPress!() : null,
+      borderRadius: BorderRadius.circular(LemonRadius.small),
+      child: Container(
+        decoration: BoxDecoration(
+          color: color,
+          borderRadius: BorderRadius.circular(LemonRadius.small),
+        ),
+        padding: EdgeInsets.symmetric(
+          horizontal: Spacing.superExtraSmall,
+          vertical: Spacing.superExtraSmall / 2,
+        ),
+        child: _buildContent(),
+      ),
+    );
+  }
+
+  Widget _buildContent() {
+    var renderKey = Characters(reactionKey!);
```

Comment:
Boilerplate codes, put it into main `build` class. We don't need to separate it unless it very complicate that need to be separate